### ISSUE TITLE
Components.default.filePreviewVC not setting my custom filePreviewVC

### ISSUE THREAD

**Issue by tciuro** [GetStream/stream-chat-swiftui#463]
## What did you do?

This is my custom filePreviewVC:
```swift
import SwiftUI
import StreamChatUI
import StreamChat

class CustomChatAttachmentPreviewVC: ChatMessageAttachmentPreviewVC {
    private lazy var shareButton = UIBarButtonItem(
        image: appearance.images.share,
        style: .plain,
        target: self,
        action: #selector(share)
    )
    
    override func setUpAppearance() {
        super.setUpAppearance()
        navigationItem.rightBarButtonItems?.append(shareButton)
    }
    
    override func updateContent() {
        super.updateContent()
        title = content?.lastPathComponent
    }
    
    @objc open func share() {
        guard let content = content else {
            log.error("Can't get `attachment` to display share sheet.")
            return
        }
        
        activityIndicatorView.startAnimating()
        
        guard let pdfData = try? Data(contentsOf: content) else {
            log.error("Could not download the file to display the share sheet")
            return
        }
        let fileName = content.lastPathComponent
        let localPdfPath = FileManager.default.temporaryDirectory.appendingPathComponent(fileName)
        try? pdfData.write(to: localPdfPath)
        
        let activityViewController = UIActivityViewController(
            activityItems: [localPdfPath],
            applicationActivities: nil
        )
        activityViewController.popoverPresentationController?.barButtonItem = shareButton
        
        activityIndicatorView.stopAnimating()
        present(activityViewController, animated: true)
    }
}
```

I set it here:
```swift
    func application(
        _ application: UIApplication,
        didFinishLaunchingWithOptions _: [UIApplication.LaunchOptionsKey: Any]?
    ) -> Bool {
        Components.default.filePreviewVC = CustomChatAttachmentPreviewVC.self
        print(Components.default.filePreviewVC)

        return true
    }
```
The output of `print(Components.default.filePreviewVC)` shows `CustomChatAttachmentPreviewVC`.

## What did you expect to happen?

When I tap a PDF attachment in my conversation/message, I was expecting that it would show `CustomChatAttachmentPreviewVC`, not the default `ChatMessageAttachmentPreviewVC`.

## What happened instead?

The default `ChatMessageAttachmentPreviewVC` is shown instead.

## GetStream Environment
**GetStream Chat version:**4.50.0
**GetStream Chat frameworks:** StreamChat, StreamChatSwiftUI 4.50.1
**iOS version:**17.4
**Swift version:**swift-driver version: 1.87.3 Apple Swift version 5.9.2 (swiftlang-5.9.2.2.56 clang-1500.1.0.2.5)
**Xcode version:**15.2
**Device:**iPhone 15 Pro Simulator

**Comment by martinmitrevski** [comment]
Hi @tciuro,

Which channel view and message list do you use? Is it `ChatChannelView` from our SwiftUI SDK, or `ChatChannelViewController` from our UIKit SDK?

If you use the one from the SwiftUI, the customizations are not done by the `Components` class, we use a different approach there, as described here: https://getstream.io/chat/docs/sdk/ios/swiftui/view-customizations/.

Both SDKs are developed with their native UI framework and patterns in mind, so they have different ways of customization. That's one reason why we recommend to use only one in your project. Also, they both have the same features, so there's no need to use only one of them.

May I know what's your use-case? Why do you need both?

Best, Martin

**Comment by tciuro** [comment]
Hi @martinmitrevski!

I'm relying as much as I can on the UI provided by the `SwiftUI SDK`. I'm calling `ChatChannelListView()`.

Of course you can ask about my use-case! Happy to explain: I found out that the PDF file preview has no way (that I know of, and confirmed by tech support) to share/save the PDF. Given the type of PDF content, it's really difficult to read on an iPhone. So I want to create my own file preview VC and allow to incorporate this functionality. Of course, this would all go away the day that the `SwiftUI SDK` provides this capability.

Knowing know what I'm after, do you recommend following the link you posted earlier?

Thank you!

**Comment by martinmitrevski** [comment]
Thanks for providing additional details. In your case, you don't need to use the UIKit SDK to accomplish what you need. The link I shared above is the way to customize views in the SwiftUI SDK.

There are 2 options, depending on what you want to accomplish.
- If you want to change the default file attachments implementation (which shows a webview for all file attachments), you need to implement the following method:
```swift
func makeFileAttachmentView(
        for message: ChatMessage,
        isFirst: Bool,
        availableWidth: CGFloat,
        scrolledId: Binding<String?>
) -> some View
```
This method is called whenever a file attachment is shown on screen. You can customize its UI (or reuse our existing implementation), and you can present a different sheet than the one we present. The default implementation can be found here: https://github.com/GetStream/stream-chat-swiftui/blob/main/Sources/StreamChatSwiftUI/ChatChannel/MessageList/FileAttachmentView.swift.
- You can also implement the custom attachments, as you suggested. In this case, you can define a special attachment type for PDFs. You can follow our custom attachments sample that shows how to define a custom type, its UI and how to modify the composer to allow custom uploading of attachments: https://getstream.io/chat/docs/sdk/ios/swiftui/chat-channel-components/message-composer/.

In both cases, if your viewer is already in UIKit, you can wrap it in a view representable to display it in SwiftUI (we do the same thing with the web view).

In a nutshell, the first one is simpler, and just modifies the message cell and sheet that display a file attachment. The second one provides you more control, you can also modify the composer with it.

Hope that helps.

**Comment by tciuro** [comment]
Aha! That makes more sense now.

Sorry if I haven't been more clear about what I want to override (trying to learn the StreamChat lingo). This is the sheet, which gets displayed once I tap on the PDF attachment of the message:


<img width="350" alt="image" src="https://github.com/GetStream/stream-chat-swiftui/assets/412899/b29dd1cd-bfe4-4101-a2de-3d1aa1044a10">


I believe I would have to override the following function call?:
```swift
    associatedtype CustomAttachmentPreviewViewType: View
    /// Creates a custom attachment view shown in the preview in the composer input.
    /// - Parameters:
    ///  - addedCustomAttachments: list of already added custom attachments.
    ///  - onCustomAttachmentTap: called when a custom attachment is tapped.
    /// - Returns: view shown in the preview slot for custom composer input.
    func makeCustomAttachmentPreviewView(
        addedCustomAttachments: [CustomAttachment],
        onCustomAttachmentTap: @escaping (CustomAttachment) -> Void
    ) -> CustomAttachmentPreviewViewType
```

**Comment by tciuro** [comment]
@martinmitrevski got it working! I had to pull some of the utils and supporting functions out to my own space and works great! Now, I'll add the sharing sheet capability. Thanks so much for the help!

**Comment by martinmitrevski** [comment]
awesome, glad to hear that 🎉
